### PR TITLE
[manuf] Add CW340 exec_env to CP/FT provisioning tests

### PR DESCRIPTION
**Pull request by tommychiu-github**
Link both CW310 and CW340 to the same definition for the tests.

### CODE REVIEW

File: sw/device/silicon_creator/manuf/skus/earlgrey_a0/sival_bringup/BUILD [hunk 1]
```diff
@@ -4,6 +4,7 @@
 
 load(
     "//rules/opentitan:defs.bzl",
+    "EARLGREY_CW340_TEST_ENVS",
```
Comment (timothytrippel) [inline sw/device/silicon_creator/manuf/skus/earlgrey_a0/sival_bringup/BUILD:7]: I don't think this is necessary. Can you delete it?

**Review by timothytrippel — APPROVED** [review]

**Review by pamaury — APPROVED** [review]

**Comment by timothytrippel** [comment]
@tommychiu-github looks like this has gotten a bit stale. Could you rebase your changes on master (fix conflicts) and repush this?

**Comment by timothytrippel** [comment]
This fixes #22971.

**Comment by timothytrippel** [comment]
@tommychiu-github we follow a rebase and merge flow (no merge commits: https://opentitan.org/book/doc/contributing/github_notes.html?search=#working-in-your-local-repo), can you squash you merge commit into the single commit and re force push your branch?

**Comment by timothytrippel** [comment]
@tommychiu-github it looks like every time you re-push this PR, you are adding a commit. the goal is to squash your commits to one commit. There are many ways to accomplish this, but what I find most intuitive is an interactive rebase. To do this, you should do the following on your branch that we are trying to fix:

1. run `git rebase -i HEAD~5`, which will bring up an editor
2. for the last 3 commits in the list of 5 commits you will see, change the first word `pick` to `s` (for "squash"), make sure to keep the word `pick` next to the first two commits, as you want to keep these
3. save and quit your editor; a new editor will be opened
4. amend the commit message to remove the commit messages of the commits your are squashing,
5. save and quit your editor
6. force push your branch to update this PR with `git push origin <branch name> --force`

**Comment by tommychiu-github** [comment]
Move to new_cw340 branch

**Pull request closed without merging**